README.md

```diff
@@ -36,6 +36,34 @@ Other test tasks can by run by passing the task through the test script, e.g.:
 ./bin/test "testOnly com.mozilla.telemetry.statistics.StatsTest"
 ```
 
+### Integrating test builds locally
+
+When making changes, you should test them locally in a dependent project before
+publishing a new SNAPSHOT version to S3, since any breaking changes will be
+automatically integrated into downstream jobs on next build.
+
+To integrate your changes, you first must publish them to your local ivy repository:
+
+```
+sbt publishLocal
+```
+
+This will populate files under
+`~/.ivy2/local/com.mozilla.telemetry/moztelemetry_2.11`.
+
+All `sbt` projects by default are configured to have the local ivy repository as
+the first resolver, so your locally published artifacts _should_ take precedence
+over the snapshots in S3. So, any subsequent local build of `telemetry-streaming`,
+`telemetry-batch-view`, etc. should pull in your locally published snapshot.
+
+To test that your changes haven't broken `telemetry-streaming`, for example,
+run its test suite:
+
+```
+cd telemetry-streaming/
+sbt test
+```
```

Comment:
Should this be more forceful? Like, should we explicitly state that any PR that changes code in this repo should pass `telemetry-batch-view` and `telemetry-streaming` test suites?

Reply:
Yes, I think it makes sense to test them both directly.

Can we actually add this to CI? Is that crazy?

Reply:
> Can we actually add this to CI? Is that crazy?

The thought had crossed my mind too. I think it's doable, but would be some work and would be dubiously maintainable. In trying to reduce the possibility of changes here accidentally breaking t-b-v builds, we would be introducing the possibility of a change to t-b-v test suite running unexpectedly causing builds of moztelemetry to fail.

I would much prefer to see us change this repo to no longer publish mutable SNAPSHOT jars, and require t-b-v and other projects to opt-in to new versions by updating their dependency lists.

Reply:
> and often adds another layer of investigation to possible issues in either t-b-v or t-s (which version of this lib was it running on?)

My experience has been the inverse of this (which commit was the snapshot built on when this code was running?) 🤣 

For the time being, I think this PR matches the scope of the bug and we should break out the discussion on dependency management. We can make a bug for it, but I also have intentions of writing up a more comprehensive proposal document to facilitate a  discussion on the team of what are options are for managing dependencies between projects and make a decision about how we want to move forward and what we want to document.

Reply:
Heh, you're not wrong.

That sounds good to me. Relevant is of course @akkomar's investigation of a monorepo. We could even simplify and instead of "batch jobs" and "streaming jobs" just have.... "jobs", which means we wouldn't even have to deal with monorepo problems, and this could just live with the code that uses it.